specs/liberty/api.rst

```diff
@@ -0,0 +1,280 @@
+      "timeout": 10,
+      "max_retries": 3,
+      "port" : 22,
+      "name": "ssh_generic"
+    }
+ 
+Monitor JSON Snippet - Ping Basic
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "type" : "PING",
+      "delay" : 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "name": "ping_generic"
+    }
+
+Monitor JSON Snippet - HTTP
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "delay": 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "type": 'HTTP',
+      "http_method": 'GET',
+      "url_path": "/healthchecks",
+      "receive_string" "SUCCESS",
+      "expected_codes": [
+        200,
+        202
+      ],
+      "name": "http-generic",
+      "port" : 80
+    }
+
+Monitor JSON Snippet - HTTPS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ .. code-block:: json
+
+    {
+      "id" : "f9fc6210-8efa-4492-9b4f-d87ba98f35b8"
+      "delay": 20,
+      "timeout": 10,
+      "max_retries": 3,
+      "type": 'HTTPS',
+      "insecure": False,
+      "http_method": 'POST',
+      "payload": {
+        "content_type" : "application/json",
+        "content" "{'demo': 'JSON', 'Blob': 'of data'}"
+        }
+      },
+      "ca_cert": "<CA_CERT_BLOB>"
+      "url_path": "/healthchecks",
+      "receive_string" "SUCCESS",
+      "expected_codes": [
+        200,
+        202
+      ],
+      "name": "https-generic",
+      "port" : 443
+    }
+
+Endpoints
+=========
+
+All of these endpoints will have the usual CRUD methods available
+
+/v0.1/gslbs
+-----------
+
+Returns a list of GLSBs
+
+/v0.1/gslbs/<uuid>
+------------------
+
+Returns a GLSB
+
+/v0.1/gslbs/<uuid>/pools
+------------------------
+
+Returns GLSB's list of pools
+
+/v0.1/gslbs/<uuid>/history
+--------------------------
+
+Returns GLSB's history (up/down/degraded etc)
+
+.. note:: This may or may not make MVP. This should only be a limited history, and for MVP my just be a "last updated" field
+
+/v0.1/pools
+-----------
+
+Returns a list of Pools
+
+/v0.1/pools/<uuid>
+------------------
+
+Returns a pool
+
+/v0.1/pools/<uuid>/monitors
+---------------------------
+
+Returns a pool's list of monitors
+
+/v0.1/pools/<uuid>/pool_members
+-------------------------------
+
+Returns a pool's list of members
+
+/v0.1/pools/<uuid>/pool_members/<uuid>
+--------------------------------------
+
+Returns a pool member
+
+/v0.1/pools/<uuid>/pool_members/<uuid>/status
+---------------------------------------------
+
+Returns a pool members status (Up/Down etc)
+
+/v0.1/pools/<uuid>/pool_members/<uuid>/monitors
+-----------------------------------------------
+
+Returns a members list of monitors
+
+/v0.1/pools/<uuid>/status
+-------------------------
+
+Returns the status of a pool
+
+/v0.1/monitors
+--------------
+
+Returns a list of health monitors
+
+/v0.1/monitors/<uuid>
+---------------------
+
+Returns a monitor
+
```

Comment:
I think we need API's to associate /disassociate monitors to the pool members. Association can also be achieved by adding the monitor_id the pool member snippet.
Something like following would work:

POST /lbs/<uuid>/pool_members/<uuid>/healthmonitors/<monitor_uuid>
Associate a health monitor to the pool member

DELETE /lbs/<uuid>/pool_members/<uuid>/healthmonitors/<monitor_uuid>
Disassociate a health monitor from the pool member